### PR TITLE
extend cache object with window.history.state value

### LINKED ISSUE #180: window.history.state is being overwritten
Hi,

While working on an app using page.js router, I've noticed that pressing the browser back button to return to a page containing a dynatable with pushState enabled, causes page.js to complain about `window.history.state.path` being undefined.

I guess this happens because Dynatable is overwritting `window.history.state`. I think the issue is in this piece of code:

``` js
cache = { dynatable: { dataset: settings.dataset } };
if (!firstPush) { cache.dynatable.scrollTop = $(window).scrollTop(); }
cacheStr = JSON.stringify(cache);

[...]

try {
    window.history[pushFunction](cache, "Dynatable state", path + params + hash);
} catch(error) {
    // Make cached records = null, so that `pop` will rerun process to retrieve records
    cache.dynatable.dataset.records = null;
    window.history[pushFunction](cache, "Dynatable state", path + params + hash);
}
```

This can be solved extending the object in `cache` variable with `window.history.state` value. Like this:

``` js
cache = { dynatable: { dataset: settings.dataset } };
if (!firstPush) { cache.dynatable.scrollTop = $(window).scrollTop(); }

// added code
$.extend(cache, window.history.state);
// end of added code

cacheStr = JSON.stringify(cache);
```

What do you think?

### PR DESCRIPTION
**Pull request by hugoduraes**
Fixes #180

### CODE REVIEW

File: jquery.dynatable.js [hunk 1]
```diff
@@ -857,6 +857,7 @@
 
       cache = { dynatable: { dataset: settings.dataset } };
       if (!firstPush) { cache.dynatable.scrollTop = $(window).scrollTop(); }
+      $.extend(cache, window.history.state);
```
Comment (JangoSteve) [inline jquery.dynatable.js]: I was initially worried that this might introduce a bug, e.g. I had previously queried by dynatable table with a search string, and then I deleted the search string and hit enter, which now adds a new state to the pushstate history that has no query, whereas the previous state had a query; I was worried that calling extend could potentially merge the old state with the new for dynatable, but because we're only doing a shallow copy with `$.extend` above and because all of the dynatable settings are namespaced under the `dynatable` key in the cache object, the new empty dynatable settings object would overwrite the old populated dynatable object.

In short, I think we're good to go here.

Comment (hugoduraes) [inline jquery.dynatable.js]: Nice to hear that! :+1:

**Comment by hugoduraes** [comment]
It looks this repo is not maintained anymore and my OCD is not letting me be comfortable with this pull request opened for so long on my pull requests list. Closing it!
